docs/advanced-topics.adoc

```diff
@@ -0,0 +1,41 @@
+ifdef::env-github,env-browser[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+:outfilesuffix: .adoc
+endif::[]
+
+== Advanced Topics
+
+=== Debugging
+
+If you are struggling with running your Arquillian test and you need more info about what is going on underneath, then you can specify `-Darquillian.debug=true`. Arquillian will print out its lifecycle so you will know exactly where it got stuck.
+The output has four types of information - you can distinguish them by their prefix:
+
+* (E) Arquillian event that was fired
+* (I) Arquillian interceptor that observed the related event context
+* (O) Arquillian observer that observed the related event
+* (X) Arquillian extension that was registered
+
+
+=== Test Observer
+
+In case you need to observe one or more Arquillian events and you don't want to deal with the whole SPI machinery, you can use the annotation `@Observer` at your test class and observe any Arquillian event in the context of the related test class (starting with `BeforeClass` and ending with `AfterClass`):
+
```

Comment:
Maybe worth to distinguish here that with SPI it is global for all ARQ tests, whereas here it's more for test with some specific needs. I think this value proposition is even better than freeing user from dealing with `SPI`. That said - let's rephrase first paragraph (bonus points: re-use in the blog announcement)